src/ObjectMultiplex.ts

```diff
@@ -0,0 +1,93 @@
+import { Duplex } from 'readable-stream';
+import eos from 'end-of-stream';
+import once from 'once';
+import { Substream } from './Substream';
+
+const IGNORE_SUBSTREAM: Record<string, unknown> = {};
+
+export class ObjectMultiplex extends Duplex {
+
+  private _substreams: Record<string, Substream | Record<string, any>>;
+
+  constructor(_opts: Record<string, unknown> = {}) {
+    const opts = Object.assign({}, _opts, {
+      objectMode: true,
+    });
+    super(opts);
+    this._substreams = {};
+  }
+
+  createStream(name: string): Substream {
+    // validate name
+    if (!name) {
+      throw new Error('ObjectMultiplex - name must not be empty');
+    }
+
+    if (this._substreams[name]) {
+      throw new Error(`ObjectMultiplex - Substream for name "${name}" already exists`);
+    }
+
+    // create substream
+    const substream = new Substream({ parent: this, name });
+    this._substreams[name] = substream;
+
+    // listen for parent stream to end
+    anyStreamEnd(this, (_error?: Error | null) => {
+      substream.destroy();
+    });
+
+    return substream;
+  }
+
+  // ignore streams (dont display orphaned data warning)
+  ignoreStream(name: string): void {
+    // validate name
+    if (!name) {
+      throw new Error('ObjectMultiplex - name must not be empty');
+    }
+    if (this._substreams[name]) {
+      throw new Error(`ObjectMultiplex - Substream for name "${name}" already exists`);
+    }
+    // set
+    this._substreams[name] = IGNORE_SUBSTREAM;
+  }
+
+  _read(): void {
+    return undefined;
+  }
+
+  _write(
+    chunk: Record<string, any>,
```

Comment:
We should create an interface for `chunk`, probably just called `Chunk`.

The type of `Chunk.data` should be `unknown` if that's manageable. If not, it can be `any`.